js/ui/animation.js

```diff
@@ -99,17 +116,31 @@ var Animation = new Lang.Class({
         this.stop();
     },
 
-    setFrameTimeouts: function(timeoutList) {
-        if (timeoutList.length == 0)
-            throw new Error('Cannot set an empty list as the frame-timeouts for animation!');
+    _parseFrame: function(frameInfo) {
+        if (typeof frameInfo === 'string') {
+            let [frame, delay] = frameInfo.split(' ');
+            frame = parseInt(frame);
 
+            if (delay.indexOf('-') > -1)
+                delay = parseInt(delay);
+
+            return [frame, delay];
+        }
+        return [frameInfo, this._speed];
+    },
+
+    setFrames: function(framesList) {
```

Comment:
Maybe call it something more detailed like `setFrameInfo` or `setFrameMetadata` (like there was before: `setFrameTimeouts`).

Reply:
+1